Assets/Scripts/UI/CoreGameplay/Carousel/Carousel.cs

```diff
@@ -204,9 +206,37 @@ private void ClearHighlightOnItem(CarouselItem itemToClear)
         itemToClear.GetComponent<Image>().color = unSelectionColour;
     }
 
+    public void UpdateCarouselInstrumentImages(Musician musician)
+    {
+        if (carouselType != CarouselType.Instrument) 
+        {
+            StSDebug.LogWarning("Something went wrong and this is not an instrument carousel");
+            return;
+        }
+
+        foreach (var item in _contentItems)
+        {
+            Instrument instrument = item.item as Instrument;
+            if (instrument == null)
+            {
+                continue;
+            }
+            if (musician != null)
```

Comment:
Could this null check be moved to the start of the function instead?

Reply:
I think not, at least it needs a check through every loop iteration anyways